_includes/article.html

```diff
@@ -25,7 +25,7 @@
 
 {% macro render(post, path) %}
 
-{% from "macros.html" import share as share %}
+{% from "macros.html" import share as share with context %}
```

Comment:
that's the only good way to make sure the request object propagates, which the share code uses.